src/query/sql/src/planner/optimizer/rule/rewrite/rule_eager_aggregation.rs

```diff
@@ -0,0 +1,813 @@
+                    } else {
+                        join_expr.replace_children(vec![
+                            SExpr::create_unary(
+                                RelOperator::Aggregate(eager_count_agg_final),
+                                SExpr::create_unary(
+                                    RelOperator::Aggregate(eager_count_agg_partial),
+                                    join_expr.child(0)?.clone(),
+                                ),
+                            ),
+                            SExpr::create_unary(
+                                RelOperator::Aggregate(eager_agg_final),
+                                SExpr::create_unary(
+                                    RelOperator::Aggregate(eager_agg_partial),
+                                    join_expr.child(1)?.clone(),
+                                ),
+                            ),
+                        ])
+                    }
+                }
+                false => {
+                    if d == 0 {
+                        join_expr.replace_children(vec![
+                            SExpr::create_unary(
+                                RelOperator::Aggregate(eager_agg_final),
+                                SExpr::create_unary(
+                                    RelOperator::Aggregate(eager_agg_partial),
+                                    join_expr.child(0)?.clone(),
+                                ),
+                            ),
+                            join_expr.child(1)?.clone(),
+                        ])
+                    } else {
+                        join_expr.replace_children(vec![
+                            join_expr.child(0)?.clone(),
+                            SExpr::create_unary(
+                                RelOperator::Aggregate(eager_agg_final),
+                                SExpr::create_unary(
+                                    RelOperator::Aggregate(eager_agg_partial),
+                                    join_expr.child(1)?.clone(),
+                                ),
+                            ),
+                        ])
+                    }
+                }
+            };
+
+            let mut result = eval_scalar_expr
+                .replace_children(vec![
+                    final_agg_final_expr
+                        .replace_children(vec![
+                            final_agg_partial_expr
+                                .replace_children(vec![join_expr])
+                                .replace_plan(final_agg_partial.try_into()?),
+                        ])
+                        .replace_plan(final_agg_final.try_into()?),
+                ])
+                .replace_plan(eval_scalar.try_into()?);
+
+            result.set_applied_rule(&self.id);
+            state.add_result(result);
+        }
+        Ok(())
+    }
+
+    fn pattern(&self) -> &SExpr {
+        &self.pattern
+    }
+}
+
+fn get_columns_set(s_expr: &SExpr, columns_set: &mut ColumnSet) {
```

Comment:
Maybe what you want is `RelationalProperty::output_columns`? Which can be constructed by:
```rust
let rel_expr = RelExpr::with_s_expr(s_expr);
let prop = rel_expr.derive_relational_prop()?;

let output_columns = prop.output_columns;
```